### ISSUE TITLE
Multiple Inferences in the same session - Cpp

### ISSUE THREAD

**Issue by pcaselles** [microsoft/onnxruntime#2700]
**Describe the bug**
I can call session.run(...) only once for each session, otherwise an error occur. It only happens with GPU enabled. The first inference of the session succesfuly works, after that the problem appears. If I create a new session for each inference, all works properly, but it's not the correct way of making multiple inferences. The same code in CPU (commenting the line for enableing GPU) succesfuly works.

**Error**
[E:onnxruntime:, sequential_executor.cc:165 onnxruntime::sequentialExecutor::Execute] Non-zero status code returned while running GatherND node.

**Additional context**
The main idea is to make multiple inferences in one session without creating it every time.

**Expected behavior**
I would expect to run succesfuly as it does in CPU mode. 

**System information**
- Windows 7
- ONNX Runtime version: 1.0
- CUDA/cuDNN version: 10.0
- C++

**To Reproduce**
Ort::Env envv(ORT_LOGGING_LEVEL_WARNING, "test");
Ort::SessionOptions ses_options = Ort::SessionOptions();
ses_options.SetIntraOpNumThreads(1);
OrtSessionOptionsAppendExecutionProvider_CUDA(ses_options, 0);

ses_options.SetGraphOptimizationLevel(GraphOptimizationLevel::ORT_ENABLE_EXTENDED);

Ort::Session sesi = Ort::Session(envv, config.path_model, ses_options);

auto mem_info = Ort::MemoryInfo::CreateCpu(OrtArenaAllocator, OrtMemTypeDefault);
inputTensor = ...
	
auto output_tensors = sesi.Run(Ort::RunOptions{ nullptr }, input_node_names.data(), inputTensor, input_node_names.size(), output_node_names.data(), config.output_node_names.size());

auto output_tensors2 = sesi.Run(Ort::RunOptions{ nullptr }, input_node_names.data(), inputTensor, input_node_names.size(), output_node_names.data(), output_node_names.size());

**Comment by hariharans29** [comment]
Can you please try this on Windows 10 ? Or you can share the model and I ll try running it with dummy data.

**Comment by pcaselles** [comment]
I've try it on windows 10 and the same problem has appeared. I've been using maskrcnn model. I generated this model from keras-onnx-master. 
Here is the link to download it: https://we.tl/t-r2Dw2bO4uZ

**Comment by hariharans29** [comment]
Thanks. I ll take a look. Could you please try with the recent 1.1 release if you can ? I ll give it a shot when I get to it as well.

**Comment by hariharans29** [comment]
@wenbingl @jiafatom @EmmaNingMS - Could you please investigate if there is a regression in keras-onnx-master wrt the Mask RCNN model ?

The model shared by the OP seem to have weird input/output shapes when compared with the Mask RCNN model shared a while ago to test against ORT.

OP's model - 

![image](https://user-images.githubusercontent.com/9969784/71384499-8c74ac00-2596-11ea-8ce2-50bf22b3a8c0.png)

Model shared by @jiafatom -

![image](https://user-images.githubusercontent.com/9969784/71384506-972f4100-2596-11ea-9a57-558a77dd557e.png)

@pcaselles - How does your `input_image` tensor look like ? From the model you shared, it is expected to be of shape [N, 0, 0, 3] - this is basically an empty tensor. I am not sure how any inferencing works honestly given that the input is expected to be empty. In your first successful run, are the output labels correct ?

![image](https://user-images.githubusercontent.com/9969784/71384556-d78ebf00-2596-11ea-898b-8f0ce6144ffb.png)

**Comment by pcaselles** [comment]
I've linked a wrong model with wrong input tensors shape, I aplogise. 

You can use the one uploaded in the modelzoo, or the one you provided in this post. It happens the same, in fact once I discovered this error, I've been using that model to be sure is well generated.

**Comment by hariharans29** [comment]
@pcaselles - AFAIK, there is no keras Mask RCNN model in the model zoo. There is a Mask RCNN (from PyTorch), could you please provide a link to the exact model to avoid confusion ?

I was able to run the Mask RCNN model a couple of times provided after conversion by the keras-onnx folks using the same session instance. So the problem might be with the specific model you are referring to. So a link would help. Thanks.

**Comment by pcaselles** [comment]
You can use this model, the exact that is giving me problems: https://we.tl/t-0jtXaTyxI3
This model is from tensorflow/keras using this: https://github.com/onnx/keras-onnx

Are you using gpu acceleration? What is more wird is that sometimes works and doesn't give any errors, but only few times.

**Comment by hariharans29** [comment]
Yes, I was using GPU acceleration. I ll try and see if I can run inferencing for about 10 times sequentially.

Is there a specific input that it errors out on ? If so, sharing that would help.

**Comment by pcaselles** [comment]
No, with different inputs always appear the same error.

**Comment by pcaselles** [comment]
I've used the new release of onnxruntime. I've seen an improvement, now the model can be inferenced in GPU, but with some issues:

After doing more than 300000 inferences we got more than 3% errors. Sometimes there are streaks, meaning that after doing 300 clean inferences suddenly almost 30  consecutive errors appear. 

The error that shows up, is the same with the previous release, however now it only appear when fails, not always.

It's a normal behaviour?
